controllers/contentlibrary/utils/utils.go

```diff
@@ -65,8 +66,24 @@ func AddContentLibraryRefToAnnotation(to client.Object, ref *imgregv1a1.NameAndK
 	if annotations == nil {
 		annotations = map[string]string{}
 	}
-	annotations[v1alpha2.VMIContentLibRefAnnotation] = string(data)
+	annotations[vmopv1.VMIContentLibRefAnnotation] = string(data)
 	to.SetAnnotations(annotations)
 
 	return nil
 }
+
+// AddImageTypeToLabel adds the image type label to the given object if the
+// label value is valid.
+func AddImageTypeToLabel(to client.Object, imageType string) error {
+	if errStrs := validation.IsValidLabelValue(imageType); errStrs != nil {
+		return fmt.Errorf("invalid image type label value %q: %v", imageType, errStrs)
```

Comment:
We really cannot return an error here. If the value isn't valid there is no point of endlessly retrying, and it will prevent the k8s object from being updated.

Either just return, set the value of an empty string, or delete the label if it exists in this case.

Reply:
Addressed in the latest commit.